File: J09999.java
public class J09999 {

    public static void main(String[] args) {
        long start = System.currentTimeMillis();
        System.out.println(solve(10));
        long end = System.currentTimeMillis();
        System.out.println("time:" + (end - start) / 1000.0 + "s");
       
        start = System.currentTimeMillis();
        System.out.println(solve(2000000));
        end = System.currentTimeMillis();
        System.out.println("time:" + (end - start) / 1000.0 + "s");
    }

    /**
     * Project Euler #10を解く
     * !Attention! このコードは遅い上、バグを含んでいます。提出には使えないので注意してください。
     *
     * @param n
     * @return 引数n以下の素数の和
     */
    public static int solve(int n) {
        n = n + 1;

        boolean[] list = new boolean[n];
        for (int i = 2; i < n; i++) {
            boolean flg = true;
            for (int j = 2; j < i; j++) {
                if (i % j == 0) {
                    flg = false;
                    break;
                }
            }
            if (flg) {
                list[i] = true;
            }
        }
        int sum = 0;
        for (int i = 2; i < n; i++) {
            if (list[i]) {
                sum += i;
            }
        }
        return sum;
    }
}

File: J09999.py
from time import time

def solve(n):
    """Project Euler #10を解く !Attention! このコードは遅いです。提出には使えないので注意してください。"""
    list_ = [0] * (n+1)
    for i in range(2,n+1):
        for j in range(2,i):
            if i % j == 0:
                break
        else:
            list_[i] = 1
    
    sum=0
    for i in range(2,n+1):
        if list_[i] == 1:
            sum += i
        
    return sum

if __name__ == '__main__':
    start = time()
    print(solve(10))
    assert 17 == solve(10)
    end = time()
    print(end - start)
    
    # 公開時は削除してください。
    assert 1060 == solve(100)
    assert 76127 == solve(1000)
    
    print('---')
    # 公開時は削除してください。

    start = time()
    num = solve(2000000)
    print(num)
    end = time()
    print(end - start)

